### PR TITLE
Suggest Fabrikate Components, Refactor Hydrate

### PR DESCRIPTION
**Pull request by andrewDoing**
This PR has a lot of changes, so bear with me. 
In hindsight, I should've split this PR into refactoring and the feature. Instead I refactored as I went with the feature.

## Refactoring
#15 progress
#10 progress, possibly done.
The structure of the project has changed from a top-level script using a library of files to a module that has a \_\_main__.py file (for running hydrate) but also will be importable as a pip package to use the functionality of hydrate in other scripts. This is the new way to run hydrate, and it's reflected in the README changes:
```bash
python -m hydrate run
```
There are more files now, they were needed to decouple the Component class, and encapsulate functionality.

## Enhancements
#13 progress, MVP functionality is probably done.
Hydrate now compares deployments on the cluster with existing Fabrikate Components that exist on GitHub. When a full match is discovered, it inserts that existing Component into the component.yaml that is generated, including the source of the component.

This is the output generated for a cluster running the cloud-native component.
```yaml
name: hydrated-cluster
generator: static
source: <source repository url>
method: git
path: ./manifests
subcomponents:
  - name: elasticsearch-fluentd-kibana
    generator: static
    source: https://github.com/microsoft/fabrikate-definitions/tree/master/definitions/fabrikate-elasticsearch-fluentd-kibana
    method: git
  - name: istio
    generator: static
    source: https://github.com/microsoft/fabrikate-definitions/tree/master/definitions/fabrikate-istio
    method: git
  - name: jaeger
    generator: static
    source: https://github.com/microsoft/fabrikate-definitions/tree/master/definitions/fabrikate-jaeger
    method: git
```

As you can see, the cloud-native component isn't reflected here. That's because I determine which Fabrikate components to use by comparing the names of the deployments on the cluster with the names of the existing components. Since cloud-native is essentially an umbrella component, hydrate cannot currently suggest it.

### CODE REVIEW

**Comment by andrewDoing** [comment]
Just realized that this PR breaks some tests. I'm going to update and add new tests on top of these changes so that this PR is actually ready to merge into master.

**Comment by andrewDoing** [comment]
Alright the tests are fixed now. Should be ready for review!

File: hydrate/scrape.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+def get_repo_components():
+    """Return the Fabrikate Component List."""
+    response = simple_get(COMP_DEFS_URL)
+    components = parse_html(response)
+    components = remove_fabrikate_prefix(components)
+    components = prepend_github_url(components)
+    return components
+
+
+def parse_html(html_bytes):
```
Comment (edaena) [inline hydrate/scrape.py]: You can also use the github contents api to get the information: https://developer.github.com/v3/repos/contents/, https://api.github.com/repos/microsoft/fabrikate-definitions/contents/definitions

Comment (andrewDoing) [reply hydrate/scrape.py]: Updated code, now it uses the api endpoint! Thanks!

File: hydrate/scrape.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    return components
+
+
+def remove_fabrikate_prefix(components):
+    """Remove the fabrikate prefix from the Component names."""
+    fabrikate_found = False
+    for component in components:
+        words = component.name.split("-")
+        new_title = ""
+        if words[0] == "fabrikate":
```
Comment (edaena) [inline hydrate/scrape.py]: You can also use a regex: name = re.sub('^fabrikate-', '', name)
More info here: https://docs.python.org/3/library/re.html

Comment (andrewDoing) [reply hydrate/scrape.py]: Nice, this reduced the size of the function significantly!

**Review by edaena — APPROVED** [review]

**Pull request merged**